src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -289,6 +289,9 @@ public static Pod combine(Pod parent, Pod template) {
                 ? parent.getSpec().getSchedulerName()
                 : template.getSpec().getSchedulerName();
 
+        Long activeDeadlineSeconds = template.getSpec().getActiveDeadlineSeconds() != null
```

Comment:
In case the template does not set the value (explicitly), but thee parent does, the parent value wins, overriding the "empty" value from the template.
But this will break "Value should be a positive integer, default being empty which means no activeDeadlineSeconds will be specified." because in the template.

Reply:
The main purpose of the parent template is to set empty fields that are not defined in the child, isn't it? If the parent template has set some kind of deadline for the Kubernetes pod, the child should follow with it or increase/decrease it if necessary, but not discard.

It might be nice to have some special value where activeDeadlineSeconds should set the default value and ignore the parent template. But creating a second Pod Template will be much easier 😃

Reply:
Anyway in the version before this commit, your problem was not solved. Earlier you always used the activeDeadlineSeconds value from the parent template, which always "won" the child yaml and you was not be able to redefine this. But now you can set your own value and it'll be change in combined template.

Reply:
I agree on the main purpose, but leaving the child empty to override a given parent value is a valid case.
But you are right, then I can create a second parent, leaving empty.

In case both parent and child have not set the value (== null), then in line 351 .withActiveDeadlineSeconds() will now be called with value null; I haven't tested it yet, will this work as expected?

Reply:
Yes, everything will work as you expected. If the child value is not set (null), the parent value will be used, but the parent value set as empty (null), so activeDeadlineSeconds will be null in the combined template